packages/analytics/amplify_analytics_pinpoint_dart/lib/src/analytics_plugin_impl.dart

```diff
@@ -323,4 +321,54 @@ class AmplifyAnalyticsPinpointDart extends AnalyticsPluginInterface {
     await _eventClient.close();
     await _eventStorageAdapter.close();
   }
+
+  /// Retrieve the stored pinpoint endpoint id
+  @visibleForTesting
+  static Future<String> retrieveEndpointId({
```

Comment:
This should be an instance method and only the app ID should be passed in (or also saved as a property on the instance). Having a static method doesn't make much sense since the method is dependent on the secure storage + legacy data provider instance variables.

Reply:
Alright, refactored the code to be an instance method instead.